### PR TITLE
Cleans up bee code very slightly, doesn't allow unsynthisisable chems to be put in bees

### PR DESCRIPTION
**Pull request by LemonInTheDark**
## About The Pull Request

Updates some documentation, makes bee synth respect the can synth flag, and implements early returns in one proc properly.

## Why It's Good For The Game

Honestly I initially did this cause of the synth thing, but at this point I care more about the early returns. Or maybe that's the nightcode talking.

Oh also I updated some documentation, yadeyada, comments do not compile all that.
This kills the tactical romerol bee

## Changelog
:cl:
balance: Bees can no longer synthesize forbidden chemicals. Looking at you romerol
/:cl:

### CODE REVIEW

File: code/modules/mob/living/simple_animal/hostile/bees.dm [hunk 1]
```diff
@@ -295,29 +295,37 @@
...
-				queen.assign_reagent(R)
-				user.visible_message("<span class='warning'>[user] injects [src]'s genome with [R.name], mutating its DNA!</span>","<span class='warning'>You inject [src]'s genome with [R.name], mutating its DNA!</span>")
-				name = queen.name
-			else
-				to_chat(user, "<span class='warning'>You don't have enough units of that chemical to modify the bee's DNA!</span>")
-	..()
+	if(!istype(I, /obj/item/reagent_containers/syringe))
+		return ..()
+
+	var/obj/item/reagent_containers/syringe/S = I
```
Comment (coiax) [inline code/modules/mob/living/simple_animal/hostile/bees.dm:301]: Under new contributing guidelines, you should use more informative variable names.

File: code/modules/mob/living/simple_animal/hostile/bees.dm [hunk 2]
```diff
@@ -295,29 +295,37 @@
...
+		user.visible_message("<span class='notice'>[user] injects [src] with royal bee jelly, causing it to split into two bees, MORE BEES!</span>","<span class='warning'>You inject [src] with royal bee jelly, causing it to split into two bees, MORE BEES!</span>")
+		return
+
+	var/datum/reagent/R = GLOB.chemical_reagents_list[S.reagents.get_master_reagent_id()]
+	if(R && S.reagents.has_reagent(R.type, 5))
+		if(!(R.chemical_flags & REAGENT_CAN_BE_SYNTHESIZED)) //Adminrazine bee man bad
+			to_chat(user, "<span class='warning'>The bee writhes in anguish, are you really going to put something so foreign inside it?</span>")
+			return
+
+		S.reagents.remove_reagent(R.type,5)
```
Comment (coiax) [inline code/modules/mob/living/simple_animal/hostile/bees.dm:322]:
```suggestion
		S.reagents.remove_reagent(R.type, 5)
```

Comment (coiax) [reply code/modules/mob/living/simple_animal/hostile/bees.dm:322]: also probably more with the variable renaming

File: code/modules/mob/living/simple_animal/hostile/bees.dm [hunk 3]
```diff
@@ -295,29 +295,37 @@
...
+		queen.assign_reagent(R)
+		user.visible_message("<span class='warning'>[user] injects [src]'s genome with [R.name], mutating its DNA!</span>","<span class='warning'>You inject [src]'s genome with [R.name], mutating its DNA!</span>")
+		name = queen.name
+		return
+		
+	to_chat(user, "<span class='warning'>You don't have enough units of that chemical to modify the bee's DNA!</span>")
```
Comment (coiax) [inline code/modules/mob/living/simple_animal/hostile/bees.dm:328]: Probably move this line to an early return, and have the "mutating bee DNA" be at the very bottom.

File: code/modules/mob/living/simple_animal/hostile/bees.dm [hunk 4]
```diff
@@ -295,29 +295,37 @@
...
+		S.reagents.remove_reagent(/datum/reagent/royal_bee_jelly, 5)
+		var/obj/item/queen_bee/qb = new(user.drop_location())
+		qb.queen = new(qb)
+		if(queen?.beegent)
+			qb.queen.assign_reagent(queen.beegent) //Bees use the global singleton instances of reagents, so we don't need to worry about one bee being deleted and her copies losing their reagents.
+		user.put_in_active_hand(qb)
+		user.visible_message("<span class='notice'>[user] injects [src] with royal bee jelly, causing it to split into two bees, MORE BEES!</span>","<span class='warning'>You inject [src] with royal bee jelly, causing it to split into two bees, MORE BEES!</span>")
+		return
+
+	var/datum/reagent/R = GLOB.chemical_reagents_list[S.reagents.get_master_reagent_id()]
```
Comment (coiax) [inline code/modules/mob/living/simple_animal/hostile/bees.dm:316]: Same here on "informative variable names"

**Review by coiax — CHANGES_REQUESTED** [review]

File: code/modules/mob/living/simple_animal/hostile/bees.dm [hunk 5]
```diff
@@ -295,29 +295,37 @@
...
+		return ..()
+
+	var/obj/item/reagent_containers/syringe/S = I
+	if(S.reagents.has_reagent(/datum/reagent/royal_bee_jelly)) //checked twice, because I really don't want royal bee jelly to be duped
+		if(!S.reagents.has_reagent(/datum/reagent/royal_bee_jelly, 5))
+			to_chat(user, "<span class='warning'>You don't have enough royal bee jelly to split a bee in two!</span>")
+			return
+
+		S.reagents.remove_reagent(/datum/reagent/royal_bee_jelly, 5)
+		var/obj/item/queen_bee/qb = new(user.drop_location())
```
Comment (Mothblocks) [inline code/modules/mob/living/simple_animal/hostile/bees.dm:308]: Under new contributing guidelines, you should use more informative variable names.

File: code/modules/reagents/chemistry/fermi_readme.md [hunk 6]
```diff
@@ -180,11 +180,12 @@ See above for purity mechanics, but this is where you set the reagents that are
 
 The flags you can set for `var/chemical_flags` are:
 ```dm 
-#define REAGENT_DEAD_PROCESS		(1<<0)	//allows on_mob_dead() if present in a dead body
-#define REAGENT_DONOTSPLIT			(1<<1)	//Do not split the chem at all during processing - ignores all purity effects
-#define REAGENT_INVISIBLE			(1<<2)	//Doesn't appear on handheld health analyzers.
-#define REAGENT_SNEAKYNAME          (1<<3)  //When inverted, the inverted chem uses the name of the original chem
-#define REAGENT_SPLITRETAINVOL      (1<<4)  //Retains initial volume of chem when splitting for purity effects 
+#define REAGENT_DEAD_PROCESS (1<<0)	//allows on_mob_dead() if present in a dead body
```
Comment (Mothblocks) [inline code/modules/reagents/chemistry/fermi_readme.md:183]:
```suggestion
#define REAGENT_DEAD_PROCESS (1<<0) //allows on_mob_dead() if present in a dead body
```

File: code/modules/reagents/chemistry/fermi_readme.md [hunk 7]
```diff
@@ -180,11 +180,12 @@ See above for purity mechanics, but this is where you set the reagents that are
...
+#define REAGENT_DONOTSPLIT (1<<1) //Do not split the chem at all during processing - ignores all purity effects
+#define REAGENT_INVISIBLE (1<<2) //Doesn't appear on handheld health analyzers.
+#define REAGENT_SNEAKYNAME (1<<3) //When inverted, the inverted chem uses the name of the original chem
+#define REAGENT_SPLITRETAINVOL (1<<4)  //Retains initial volume of chem when splitting for purity effects
```
Comment (Mothblocks) [inline code/modules/reagents/chemistry/fermi_readme.md:187]:
```suggestion
#define REAGENT_SPLITRETAINVOL (1<<4) //Retains initial volume of chem when splitting for purity effects 
```

**Comment by TemporalOroboros** [comment]
And so the hugboxing begins.

**Comment by Fikou** [comment]
hugbox BOOOOOOOOOOOOOOOOOO

**Comment by ATH1909** [comment]
NOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOO

**Comment by ATH1909** [comment]
also it's romerol

**Comment by LemonInTheDark** [comment]
> 
> 
> also, it's spelled "romerol"

Thank you

**Comment by LemonInTheDark** [comment]
> 
> 
> NOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOO
> 
> I don't suppose it'd be possible to just make bees not produce can_synth = FALSE chems in honey combs, while still allowing them to sting people with them? I like nanomachine bees...

I mean feel free to pr that, I'm mostly doing this cause of the inconsistency, if something can't be synthesized it shouldn't be able to be mass produced or distributed. I wouldn't even block it, like I said most of the reason I made this pr is the code stinking so hard

**Comment by LemonInTheDark** [comment]
Ew, the bee's gone stale

**Pull request closed without merging**